tests/test_fields.py

```diff
@@ -2082,6 +2086,29 @@ def test_against_partial_and_full_updates(self):
         field.partial = True
         assert field.get_value(QueryDict('')) == rest_framework.fields.empty
 
+    def test_valid_inputs_is_json_serializable(self):
+        for input_value, _ in get_items(self.valid_inputs):
+            validated = self.field.run_validation(input_value)
+
+            try:
+                json.dumps(validated)
+            except TypeError as e:
+                assert (
+                    False
+                ), f'Validated output not JSON serializable: {repr(validated)}; Error: {e}'
```

Comment:
Maybe a bit more explicit to achieve this would be [with `pytest.fail`](https://docs.pytest.org/en/7.1.x/reference/reference.html?highlight=fail#pytest.fail):

```suggestion
                pytest.fail(
                    f'Validated output not JSON serializable: {repr(validated)}; Error: {e}'
                )
```